### PR TITLE
Migration to PyVO for MAST TAP notebooks

### PR DESCRIPTION
**Pull request by tsdower**
As the ESA TAP/TAPPlus library is being superceded by increased support for PyVO, we are moving MAST's notebooks previously using the TAP+ client functionality to PyVO's TAP client support. New notebooks from this point will be designed with PyVO in mind. 

Development outside these notebooks is still required to clear spurious warnings, particularly regarding units, which also produced similar warnings in the TAP+ system. Once these are handled, the warning 'ignore' can be removed.

### CODE REVIEW

**Comment by theresadower** [comment]
@ceb8: Spoke to @eteq, can you review this as you're more familiar with ASB materials?
In general, we noted: 
 * async pyvo is slower than the async TAP+ calls we had before, but not slower enough to fail CI
 * there are ignored warnings from pyvo, particularly about units. some of these are the same as we had from TAP+. Some will need addressed in pyvo.
 * there will be resources dedicated to improving both of these as part of the general pyvo development.

File: notebooks/MAST/PanSTARRS/PS1_DR2_TAP/PS1_DR2_TAP.ipynb [hunk 1]
```diff
@@ -38,7 +38,7 @@
    "source": [
     "Table Access Protocol (TAP) services allow more direct and flexible access to astronomical data than the simpler types of IVOA standard data services. Queries are built with the SQL-like Astronomical Data Query Language (ADQL), and can include geographic / spatial queries as well as filtering on other characteristics of the data. This also allows the user fine-grained control over the returned columns, unlike the fixed set of coumns returned from cone, image, and spectral services.\n",
     "\n",
-    "For this example, we'll be using the astroquery TAP/TAP+ client, which was developed by the ESAC Space Data Centre for working with the GAIA catalog, but is interoperable with other valid TAP services, including those at MAST. As an astroquery project, TAP+ documentation is available at ReadTheDocs: http://astroquery.readthedocs.io/en/latest/utils/tap.html\n",
+    "For this example, we'll be using the astropy affiliated PyVO client, which is interoperable with other valid TAP services, including those at MAST. PyVO documentation is available at ReadTheDocs: https://readthedocs.org/projects/pyvo/\n",
```
Comment (ceb8) [inline notebooks/MAST/PanSTARRS/PS1_DR2_TAP/PS1_DR2_TAP.ipynb]: https://readthedocs.org/projects/pyvo/ -> https://pyvo.readthedocs.io

File: notebooks/MAST/PanSTARRS/PS1_DR2_TAP/PS1_DR2_TAP.ipynb [hunk 2]
```diff
@@ -38,7 +38,7 @@
...
     "\n",
     "We'll be using TAP+ to call the TAP service at MAST serving PanSTARRS 1 Data Release 2, now with individual detection information. The schema is described within the service, and we'll show how to inspect it. The schema is also the same as the one available via the CasJobs interface, with an additional view added for the most common positional queries. CasJobs has its own copy of the schema documentation, which can be accessed through its own site: http://mastweb.stsci.edu/ps1casjobs/\n",
```
Comment (ceb8) [inline notebooks/MAST/PanSTARRS/PS1_DR2_TAP/PS1_DR2_TAP.ipynb]: TAP+ -> PyVO

File: notebooks/MAST/PanSTARRS/PS1_DR2_TAP/PS1_DR2_TAP.ipynb [hunk 3]
```diff
@@ -109,7 +114,8 @@
    "metadata": {},
```
Comment (ceb8) [inline notebooks/MAST/PanSTARRS/PS1_DR2_TAP/PS1_DR2_TAP.ipynb]: Comment for line 103/108 (can't comment on that line):
TapPlus -> PyVO

File: notebooks/MAST/PanSTARRS/PS1_DR2_TAP/PS1_DR2_TAP.ipynb [hunk 4]
```diff
@@ -199,8 +208,8 @@
     "\"\"\".format(ra,dec,radius)\n",
```
Comment (ceb8) [inline notebooks/MAST/PanSTARRS/PS1_DR2_TAP/PS1_DR2_TAP.ipynb]: Comment for line 188/197:
`Mast._resolve_object(objname)` -> `Mast.resolve_object(objname)`
(Astroquery changed the name of this function)

File: notebooks/MAST/PanSTARRS/PS1_DR2_TAP/PS1_DR2_TAP.ipynb [hunk 5]
```diff
@@ -491,8 +500,8 @@
...
```
Comment (ceb8) [inline notebooks/MAST/PanSTARRS/PS1_DR2_TAP/PS1_DR2_TAP.ipynb]: Comment for line 481/490:
`Mast._resolve_object(objname)` -> `Mast.resolve_object(objname)`

File: notebooks/MAST/PanSTARRS/PS1_DR2_TAP/PS1_DR2_TAP.ipynb [hunk 6]
```diff
@@ -635,8 +644,8 @@
...
```
Comment (ceb8) [inline notebooks/MAST/PanSTARRS/PS1_DR2_TAP/PS1_DR2_TAP.ipynb]: Comment for line 625/634:
`Mast._resolve_object(objname)` -> `Mast.resolve_object(objname)`

File: notebooks/MAST/TESS/beginner_tess_tap_search/beginner_tess_tap_search.ipynb [hunk 7]
```diff
@@ -41,10 +41,9 @@
...
     "\n",
-    "For this example, we'll be using the astroquery TAP/TAP+ client, which was developed by the ESAC Space Data Centre for working with the GAIA catalog, but is interoperable with other valid TAP services, including those at MAST. As an astroquery project, TAP+ documentation is available at ReadTheDocs: http://astroquery.readthedocs.io/en/latest/utils/tap.html\n",
+    "For this example, we'll be using the astropy affiliated PyVO client, which is interoperable with other valid TAP services, including those at MAST. PyVO documentation is available at ReadTheDocs: https://readthedocs.org/projects/pyvo/\n",
     "\n",
-    "We'll be using TAP+ to call the CAOM Catalog TAP service at MAST and filter the results for TESS-related information. The schema for this catalog is an IVOA standard, and is also described within the service itself.\n",
-    "\n"
+    "We'll be using TAP+ to call the CAOM Catalog TAP service at MAST and filter the results for TESS-related information. The schema for this catalog is an IVOA standard, and is also described within the service itself."
```
Comment (ceb8) [inline notebooks/MAST/TESS/beginner_tess_tap_search/beginner_tess_tap_search.ipynb]: TAP+ -> PyVO

File: notebooks/MAST/TESS/beginner_tess_tap_search/beginner_tess_tap_search.ipynb [hunk 8]
```diff
@@ -110,7 +115,29 @@
    "metadata": {},
```
Comment (ceb8) [inline notebooks/MAST/TESS/beginner_tess_tap_search/beginner_tess_tap_search.ipynb]: Comment for line 104/109: 
TapPlus -> PyVO

**Review by ceb8 — CHANGES_REQUESTED** [review]
There are a number of TESS lc fits files in the TESS notebook directory. Does the notebook require them or are they files produced by the notebook that accidentally got checked in?

Overall looks great, just a few comments, mostly places where you still refer to tap+. Also, I recommend updating your version of astroquery.

**Comment by ceb8** [comment]
Oh wait, ignore what I said about the extra files. I was looking at my local directory when I thought I was looking at github. Sorry! 😛

File: notebooks/MAST/HSC/HSC_TAP/HSC_TAP.ipynb [hunk 9]
```diff
@@ -42,9 +42,9 @@
    "source": [
     "Table Access Protocol (TAP) services allow more direct and flexible access to astronomical data than the simpler types of IVOA standard data services. Queries are built with the SQL-like Astronomical Data Query Language (ADQL), and can include geographic / spatial queries as well as filtering on other characteristics of the data. This also allows the user fine-grained control over the returned columns, unlike the fixed set of coumns retunred from cone, image, and spectral services.\n",
     "\n",
-    "For this example, we'll be using the astroquery TAP/TAP+ client, which was developed by the ESAC Space Data Centre for working with the GAIA catalog, but is interoperable with other valid TAP services, including those at MAST. As an astroquery project, TAP+ documentation is available at ReadTheDocs: http://astroquery.readthedocs.io/en/latest/utils/tap.html\n",
+    "For this example, we'll be using the astropy affiliated PyVO client, which is interoperable with other valid TAP services, including those at MAST. PyVO documentation is available at ReadTheDocs: https://readthedocs.org/projects/pyvo/\n",
```
[the remark this answers is not in the document]
Comment (theresadower) [reply notebooks/MAST/HSC/HSC_TAP/HSC_TAP.ipynb]: Updated in this and the other notebooks, thanks.

File: notebooks/MAST/HSC/HSC_TAP/HSC_TAP.ipynb [hunk 10]
```diff
@@ -42,9 +42,9 @@
    "source": [
     "Table Access Protocol (TAP) services allow more direct and flexible access to astronomical data than the simpler types of IVOA standard data services. Queries are built with the SQL-like Astronomical Data Query Language (ADQL), and can include geographic / spatial queries as well as filtering on other characteristics of the data. This also allows the user fine-grained control over the returned columns, unlike the fixed set of coumns retunred from cone, image, and spectral services.\n",
```
[the remark this answers is not in the document]
Comment (theresadower) [reply notebooks/MAST/HSC/HSC_TAP/HSC_TAP.ipynb]: Fixed

File: notebooks/MAST/HSC/HSC_TAP/HSC_TAP.ipynb [hunk 11]
```diff
@@ -109,7 +111,7 @@
    "metadata": {},
```
[the remark this answers is not in the document]
Comment (theresadower) [reply notebooks/MAST/HSC/HSC_TAP/HSC_TAP.ipynb]: Whoops, fixed this in this and the other notebooks, thanks.

File: notebooks/MAST/PanSTARRS/PS1_DR2_TAP/PS1_DR2_TAP.ipynb (hunk 4 above)
Comment (theresadower) [reply notebooks/MAST/PanSTARRS/PS1_DR2_TAP/PS1_DR2_TAP.ipynb]: Updated this. The non-TAP HSC notebook may need updated as well, @rlwastro wrote that one.

**Review by ceb8 — APPROVED** [review]

File: notebooks/MAST/PanSTARRS/PS1_DR2_TAP/PS1_DR2_TAP.ipynb (hunk 4 above)
Comment (ceb8) [reply notebooks/MAST/PanSTARRS/PS1_DR2_TAP/PS1_DR2_TAP.ipynb]: I can check and do a quick update PR on that one if needed.

**Pull request merged**